src/main/scala/Dromajo.scala

```diff
@@ -0,0 +1,101 @@
+package testchipip
+
+import chisel3._
+import chisel3.util._
+import chisel3.experimental.{IntParam, StringParam}
+import freechips.rocketchip.config.{Parameters}
+
+/**
+ * Dromajo bridge to input instruction streams and check with Dromajo
+ */
+class SimDromajoBridge(insnWidths: TracedInstructionWidths, numInsns: Int) extends Module
+{
+  val io = IO(new Bundle {
+    val trace = Input(new TileTraceIO(insnWidths, numInsns))
+  })
+
+  // constants
+  val xLen = 64
+  val instBits = 32
+  // constants
+
+  val traces = io.trace.insns
+
+  val dromajo = Module(new SimDromajoCosimBlackBox(numInsns, xLen))
+
+  dromajo.io.clock := clock
+  dromajo.io.reset := reset.asBool
+
+  // TODO: convert to use SInts?
+  def signextPad(in: UInt, padLen: Int): UInt = {
+    require(in.getWidth <= padLen)
+    if (in.getWidth == padLen) {
+      in
+    } else {
+      Mux(in(in.getWidth - 1) === 1.U, Cat(0.U((padLen - in.getWidth).W) - 1.U, in), in.pad(padLen))
+    }
+  }
+
+  dromajo.io.valid := Cat(traces.map(t => t.valid).reverse)
+  dromajo.io.hartid := 0.U
+  dromajo.io.pc := Cat(traces.map(t => signextPad(t.iaddr, xLen)).reverse)
+  dromajo.io.inst := Cat(traces.map(t => t.insn.pad(instBits)).reverse)
+  dromajo.io.wdata := Cat(traces.map(t => signextPad(t.wdata, xLen)).reverse)
+  dromajo.io.mstatus := 0.U
```

Comment:
Will this work if `mstatus` isn't connected?

Reply:
Yup. Dromajo currently doesn't check against `mstatus`.